### ISSUE TITLE
Creation of training set data failing to produce full output (macOS)

### ISSUE THREAD

**Issue by FrancesBW** [mriener/gausspyplus#8]
For context, I am running on python 3.11.0 and had to make a few changes in the source code to replace some deprecated functions with their new versions, mainly for numpy. I wanted to run the code in python 3.5, however there wasn't a simple way to run python 3.5 as I am using a new MacbookPro with an Apple M3 processor, which from what I can gather doesn't support python versions that old.

Other relevant package versions: numpy 1.26.0 scipy 1.11.3 lmfit 1.3.2 astropy 5.3.4. Operating system macOS Sonoma 14.5

My issue arises in the creation of the training dataset. I followed the example tutorial and was able to reproduce the results without issue, so I modified the scripts from the tutorial to replace them with my HI data. I chose to use 300 spectra from the data to create the training set, however it seems that the fitting to those 300 spectra stops once it reaches 89/300 and then prints SUCCESS and saves the output. I have copied the output below:

> !python step_1-training_set__HI.py
> 
> WARNING: VerifyWarning: Invalid 'BLANK' keyword in header.  The 'BLANK' keyword is only applicable to integer data, and will be ignored in this HDU.
> 
> WARNING: UserWarning: Removing 'CROTA*' keywords from FITS header.
> 
> WARNING: FITSFixedWarning: 'datfix' made the change 'Set MJD-OBS to 58839.258446 from DATE-OBS'.
> decompose 300 spectra ...
> using 6 out of 8 cpus
>   0%|                                                   | 0/300 [00:00<?, ?it/s]
> WARNING: RuntimeWarning: Values in x were outside bounds during a minimize step, clipping to bounds
> 
> WARNING: RuntimeWarning: Values in x were outside bounds during a minimize step, clipping to bounds
> 
> WARNING: RuntimeWarning: Values in x were outside bounds during a minimize step, clipping to bounds
> 
> WARNING: RuntimeWarning: Values in x were outside bounds during a minimize step, clipping to bounds
> 
> WARNING: RuntimeWarning: Values in x were outside bounds during a minimize step, clipping to bounds
> 
> WARNING: RuntimeWarning: Values in x were outside bounds during a minimize step, clipping to bounds
> 
> WARNING: AstropyUserWarning: The fit may be unsuccessful; check fit_info['message'] for more information.
>   0%|▏                                        | 1/300 [01:11<5:58:36, 71.96s/it]
> WARNING: AstropyUserWarning: The fit may be unsuccessful; check fit_info['message'] for more information.
> 
> WARNING: AstropyUserWarning: The fit may be unsuccessful; check fit_info['message'] for more information.
>   1%|▎                                        | 2/300 [02:44<6:56:41, 83.90s/it]
> WARNING: AstropyUserWarning: The fit may be unsuccessful; check fit_info['message'] for more information.
>   4%|█▍                                      | 11/300 [10:05<3:28:03, 43.20s/it]
> WARNING: AstropyUserWarning: The fit may be unsuccessful; check fit_info['message'] for more information.
>   6%|██▌                                     | 19/300 [17:23<3:56:27, 50.49s/it]
> WARNING: AstropyUserWarning: The fit may be unsuccessful; check fit_info['message'] for more information.
>  30%|███████████▎                          | 89/300 [1:49:58<4:20:42, 74.14s/it]
> SUCCESS
> 
> 
> SAVED FILE: 'HI_gausspy-training_set_300_spectra.pickle' in 'decomposition_HI_gausspy/gpy_training'

I verified that the output only contains 89 training spectra. I tried varying the random seed, ncpus and increased the number of training spectra used from 300 to 900 and the script still output only 89 spectra. The functions all seem to terminate successfully despite some warning messages, but the output is not what it should be.
